migration/migration/012-tracing.sql

```diff
@@ -1,7 +1,158 @@
+    CREATE TYPE ps_trace.trace_id;
+
+    CREATE OR REPLACE FUNCTION ps_trace.trace_id_in(cstring)
+    RETURNS ps_trace.trace_id
+    LANGUAGE internal
+    IMMUTABLE PARALLEL SAFE STRICT
+    AS $function$uuid_in$function$;
+
+    CREATE OR REPLACE FUNCTION ps_trace.trace_id_out(ps_trace.trace_id)
+    RETURNS cstring
+    LANGUAGE internal
+    IMMUTABLE PARALLEL SAFE STRICT
+    AS $function$uuid_out$function$;
+
+    CREATE OR REPLACE FUNCTION ps_trace.trace_id_send(ps_trace.trace_id)
+    RETURNS bytea
+    LANGUAGE internal
+    IMMUTABLE PARALLEL SAFE STRICT
+    AS $function$uuid_send$function$;
+
+    CREATE OR REPLACE FUNCTION ps_trace.trace_id_recv(internal)
+    RETURNS ps_trace.trace_id
+    LANGUAGE internal
+    IMMUTABLE PARALLEL SAFE STRICT
+    AS $function$uuid_recv$function$;
+
+    CREATE OR REPLACE FUNCTION ps_trace.trace_id_ne(ps_trace.trace_id, ps_trace.trace_id)
+    RETURNS bool
+    LANGUAGE internal
+    IMMUTABLE PARALLEL SAFE STRICT
+    AS $function$uuid_ne$function$;
+
+    CREATE OR REPLACE FUNCTION ps_trace.trace_id_eq(ps_trace.trace_id, ps_trace.trace_id)
+    RETURNS bool
+    LANGUAGE internal
+    IMMUTABLE PARALLEL SAFE STRICT
+    AS $function$uuid_eq$function$;
+
+    CREATE OR REPLACE FUNCTION ps_trace.trace_id_ge(ps_trace.trace_id, ps_trace.trace_id)
+    RETURNS bool
+    LANGUAGE internal
+    IMMUTABLE PARALLEL SAFE STRICT
+    AS $function$uuid_ge$function$;
+
+    CREATE OR REPLACE FUNCTION ps_trace.trace_id_le(ps_trace.trace_id, ps_trace.trace_id)
+    RETURNS bool
+    LANGUAGE internal
+    IMMUTABLE PARALLEL SAFE STRICT
+    AS $function$uuid_le$function$;
+
+    CREATE OR REPLACE FUNCTION ps_trace.trace_id_gt(ps_trace.trace_id, ps_trace.trace_id)
+    RETURNS bool
+    LANGUAGE internal
+    IMMUTABLE PARALLEL SAFE STRICT
+    AS $function$uuid_gt$function$;
+
+    CREATE OR REPLACE FUNCTION ps_trace.trace_id_lt(ps_trace.trace_id, ps_trace.trace_id)
+    RETURNS bool
+    LANGUAGE internal
+    IMMUTABLE PARALLEL SAFE STRICT
+    AS $function$uuid_lt$function$;
+
+    CREATE OR REPLACE FUNCTION ps_trace.trace_id_cmp(ps_trace.trace_id, ps_trace.trace_id)
+    RETURNS int
+    LANGUAGE internal
+    IMMUTABLE PARALLEL SAFE STRICT
+    AS $function$uuid_cmp$function$;
+
+    CREATE OR REPLACE FUNCTION ps_trace.trace_id_hash(ps_trace.trace_id)
+    RETURNS int
+    LANGUAGE internal
+    IMMUTABLE PARALLEL SAFE STRICT
+    AS $function$uuid_hash$function$;
+
+    CREATE TYPE ps_trace.trace_id(
+        internallength = 16,
+        INPUT = ps_trace.trace_id_in,
+        OUTPUT = ps_trace.trace_id_out,
+        SEND = ps_trace.trace_id_send,
+        RECEIVE = ps_trace.trace_id_recv,
+        alignment = double
+    );
+
+    CREATE OPERATOR ps_trace.= ( 
+        FUNCTION=ps_trace.trace_id_eq,
+        LEFTARG=ps_trace.trace_id,
+        RIGHTARG=ps_trace.trace_id,
+        COMMUTATOR= OPERATOR(ps_trace.=),
+        NEGATOR= OPERATOR(ps_trace.!=),
```

Comment:
Is there a meaningful difference between `<>` and `!=`?

Reply:
I believe not